test/async-hooks/test-pipeconnectwrap.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const tick = require('./tick');
+const initHooks = require('./init-hooks');
+const { checkInvocations } = require('./hook-checks');
+
+const net = require('net');
+
+common.refreshTmpDir();
+
+const hooks = initHooks();
+hooks.enable();
+let pipe1, pipe2, pipe3;
+let pipeconnect;
+
+net.createServer(function(c) {
+  c.end();
+  this.close();
+}).listen(common.PIPE, common.mustCall(onlisten));
+
+function onlisten() {
+  let pipes = hooks.activitiesOfTypes('PIPEWRAP');
+  let pipeconnects = hooks.activitiesOfTypes('PIPECONNECTWRAP');
+  assert.strictEqual(
+    pipes.length, 1,
+    'one pipe wrap created when net server is listening');
+  assert.strictEqual(
+    pipeconnects.length, 0,
+    'no pipeconnect wrap created when net server is listening');
+
+  net.connect(common.PIPE, common.mustCall(onconnect));
+
+  pipes = hooks.activitiesOfTypes('PIPEWRAP');
+  pipeconnects = hooks.activitiesOfTypes('PIPECONNECTWRAP');
+  assert.strictEqual(pipes.length, 2,
+                     '2 pipe wraps created when connecting client');
+  assert.strictEqual(pipeconnects.length, 1,
+                     '1 connectwrap created when connecting client');
+
+  pipe1 = pipes[0];
+  pipe2 = pipes[1];
+  pipeconnect = pipeconnects[0];
+
+  assert.strictEqual(pipe1.type, 'PIPEWRAP', 'first is pipe wrap');
+  assert.strictEqual(pipe2.type, 'PIPEWRAP', 'second is pipe wrap');
+  assert.strictEqual(pipeconnect.type, 'PIPECONNECTWRAP',
+                     'third is pipeconnect wrap');
+  [ pipe1, pipe2, pipeconnect ].forEach(check);
+
+  function check(a) {
+    assert.strictEqual(typeof a.uid, 'number', 'uid is a number');
+    assert.strictEqual(typeof a.triggerId, 'number', 'triggerId is a number');
+    checkInvocations(a, { init: 1 }, 'after net.connect');
+  }
+}
+
+function onconnect() {
+  const pipes = hooks.activitiesOfTypes('PIPEWRAP');
+  const pipeconnects = hooks.activitiesOfTypes('PIPECONNECTWRAP');
+
+  assert.strictEqual(pipes.length, 3,
+                     '3 pipe wraps created when client connected');
```

Comment:
This is currently failing on CentOS, not sure why yet. Example failure: https://ci.nodejs.org/job/node-test-commit-linux/9123/nodes=centos5-64/console

Also, the error message in those failures is pretty bad, it doesn’t actually tell anything about the state of `pipes`. I would say that (and I think @sam-github said something similar elsewhere) that most of the time these extra `message` parameters for assertions are somewhere between redundant at best and actual obstacles to debugging at worst… I’d consider just dropping it in all places where they just restate the assertion.